pkg/proxy/iptables/proxier.go

```diff
@@ -1378,20 +1334,15 @@ func (proxier *Proxier) syncProxyRules() {
 			)
 		}
 
-		if !hasEndpoints {
-			continue
+		if len(clusterEndpoints) != 0 {
+			// Write rules jumping from svcChain to clusterEndpoints
+			proxier.writeServiceToEndpointRules(svcNameString, svcInfo, svcChain, clusterEndpoints, args)
```

Comment:
No, because we still need the cluster endpoints for the pod-to-LB special case. Sometimes. The logic is in `svcInfo.UsesClusterEndpoints()` now, and clusterEndpoints will be nil if that returned false, but I can just call UsesClusterEndpoint again here since that would be a little clearer